src/Uno.UI/Controls/Window.iOS.cs

```diff
@@ -108,6 +115,37 @@ public override void PressesBegan(NSSet<UIPress> presses, UIPressesEvent evt)
 						handled |= focusInputHandler.TryHandleDirectionalFocus(Windows.System.VirtualKey.Down);
 					}
 				}
+
+#if __MACCATALYST__
+				var virtualKey = VirtualKeyHelper.FromKeyCode(press.Key.KeyCode);
+
+				var args = new KeyEventArgs(
+					"keyboard",
+					virtualKey,
+					new CorePhysicalKeyStatus
+					{
+						ScanCode = (uint)press.Key.KeyCode,
+						RepeatCount = 1,
+					});
+
+				if (this.Log().IsEnabled(LogLevel.Trace))
+				{
+					this.Log().Trace($"PressesBegan: {press.Key.KeyCode} -> {virtualKey}");
+				}
+
+				try
+				{
+					if (_ownerEvents is { })
```

Comment:
Is it common that the `_ownerEvents` to be `null`? If so we should check it before creating the event args